test/active_model_serializers/register_jsonapi_renderer_test_isolated.rb

```diff
@@ -12,7 +12,9 @@ class << self
     end
 
     def render_with_jsonapi_renderer
-      author = Author.new(params[:data][:attributes])
+      unlocked_params = Rails::VERSION::MAJOR >= 5 ? params.to_unsafe_h : params
+      attributes = unlocked_params[:data].present? ? unlocked_params[:data][:attributes] : {}
```

Comment:
You are right that params[:data] is not set at all and the reason is that app doesn't know how to parse provided data. 

`.require` will not help since it doesn't see data at all. Also, require will cause problem when jsonapi is not registered since it doesn't see the actual value.